docs/docs/nlu-training-data.mdx

```diff
@@ -3,92 +3,166 @@ id: nlu-training-data
+### Regular Expressions for Intent Classification
 
+You can use regular expressions to improve intent classification by including the `RegexFeaturizer` component in your pipeline. When using the `RegexFeaturizer`, a regex does not act as a rule for classifying an intent. It only provides a feature that the intent classifier will use
+to learn patterns for intent classification.
+Currently, all intent classifiers make use of available regex features.
 
-If you are using the regular expressions to directly extract entities using the [RegexEntityExtractor](./components.mdx#regexentityextractor),
-the name of the regular expression should match the name of the entity you want to extract.
+The name of a regex in this case is a human readable description. It can help you remember what a regex is used for, and it is the title of the corresponding pattern feature. It does not have to match any intent or entity name. A regex for a "help" request might look like this:
 
-Try to create your regular expressions in a way that they match as few words as possible. E.g. using `hey[^\\s]*`
-instead of `hey.\*`, as the later one might match the whole message whereas the first one only matches a single word.
+```yaml-rasa
+nlu:
+- regex: help
+  examples: |
+    - \bhelp\b
+```
 
-When using the [RegexFeaturizer](./components.mdx#regexfeaturizer), the regex features for entity extraction are currently only supported by the
-`CRFEntityExtractor` and the `DIETClassifier` component! Hence, other entity extractors, like
-`MitieEntityExtractor` or `SpacyEntityExtractor` won't use the generated features and their
-presence will not improve entity recognition for these extractors. Currently, all intent classifiers make use of
-available regex features.
+The intent being matched could be `greet`,`help_me`, `assistance` or anything else.
 
-:::note
-Regex features only define entities when used in combination with the [RegexEntityExtractor](./components.mdx#regexentityextractor).
-Otherwise they don't define entities nor intents! They simply provide patterns to help the classifier
-recognize entities and related intents. Hence, you still need to provide intent & entity examples as part of your
-training data!
+Try to create your regular expressions in a way that they match as few
+words as possible. E.g. using `\bhelp\b` instead of `help.*`, as the
+later one might match the whole message whereas the first one only
+matches a single word.
 
+:::note Needs Training Examples
+The `RegexFeaturizer` provides features to the intent classifier, but it doesn't predict the intent directly. Include enough examples containing the regular expression so that the intent classifier can learn ot use the regular expression feature.
 :::
 
+### Regular Expressions for Entity Extraction
+
+If your entity has a deterministic structure, you can use regular expressions in one of two ways:
+
+1. As features for the [`RegexFeaturizer`](components.mdx#regexfeaturizer) component in the pipeline.
```

Comment:
Lets do these number sections as headers so that everything under it renders correctly 
```suggestion
#### 1. Regular expressions as features for the [`RegexFeaturizer`](components.mdx#regexfeaturizer) component in the pipeline.
```